### PR TITLE
Move iochannel init to after null check and update build script.

### PR DESCRIPTION
**Pull request by Hnefi**
Add updated package requirements for centos build.

### CODE REVIEW

**Comment by Hnefi** [comment]
DONT MERGE THIS.

**Comment by Hnefi** [comment]
@neo-apz  This is ready to merge when you have the time, thanks.

**Review by neo-apz — APPROVED** [review]

**Pull request merged**
